src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -603,7 +590,14 @@ public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
                 try
                 {
                     bool hasCompleteReferences = true;
+
                     var newReferences = new List<MetadataReference>();
+
+                    // we don't consider metadata reference being actually exist on disk as a part of complete reference
+                    // since we know some projects do have those but many just work fine since those dlls are not actually used.
```

Comment:
So missing DLLs would have already been filtered out by RAR; the only case we get missing DLLs are when they did something like a clean or they are P2P references. Those I think we _do_ want to consider incomplete.